src/utils/api/schema.js

```diff
@@ -39,33 +44,42 @@ const createUISchema = (fieldSchema, formDisplaySchema, widgets) =>
         ).length,
     )
     .sort((a, b) => formDisplaySchema[a].weight - formDisplaySchema[b].weight)
-    .reduce((acc, currentFieldName) => {
-      const widget =
-        widgets[
-          Object.keys(widgets)
-            .filter(name =>
-              formDisplaySchema[currentFieldName].type.startsWith(name),
-            )
-            .shift()
-        ];
-      const inputProps = {
-        ...(Object.prototype.hasOwnProperty.call(fieldSchema, currentFieldName)
-          ? fieldSchema[currentFieldName].attributes.settings
-          : {}),
-        ...(Object.prototype.hasOwnProperty.call(
-          formDisplaySchema,
-          currentFieldName,
-        )
-          ? formDisplaySchema[currentFieldName].settings
-          : {}),
-      };
-      acc.push({
-        fieldName: currentFieldName,
-        constraints: [],
-        widget,
-        inputProps,
-      });
-      return acc;
-    }, []);
+    .reduce(
+      (acc, currentFieldName) => {
+        const widget =
+          widgets[
+            Object.keys(widgets)
+              .filter(name =>
+                formDisplaySchema[currentFieldName].type.startsWith(name),
+              )
+              .shift()
+          ];
+        const inputProps = {
+          ...(Object.prototype.hasOwnProperty.call(
+            fieldSchema,
+            currentFieldName,
+          )
+            ? fieldSchema[currentFieldName].attributes.settings
+            : {}),
+          ...(Object.prototype.hasOwnProperty.call(
+            formDisplaySchema,
+            currentFieldName,
+          )
+            ? formDisplaySchema[currentFieldName].settings
+            : {}),
+        };
+        acc[
```

Comment:
I'm wondering whether we should keep the existing createUiSchema function and instead have a second function which creates the left/right information out of an existing schema. ❓